core/src/main/java/org/mule/DefaultMuleMessage.java

```diff
@@ -1322,6 +1353,24 @@ private void writeObject(ObjectOutputStream out) throws Exception
         return toWrite;
     }
 
+    @Override
+    protected Serializable serializeValue(ObjectOutputStream out) throws Exception
+    {
+        byte[] serializablePayload = muleContext.getTransformationService().getPayloadAsBytes(this);
```

Comment:
BTW: This can all go away if transform the message to byte[] earlier before serialization, but that's a separate issue/PR.  So this would potentially disappear soon.